### ISSUE TITLE
Insee API 429 error not consistent - API rate consumption not respected

### ISSUE THREAD

**Issue by tgrandje** [InseeFrLab/pynsee#221]
429 error code does not seem to be consistently return by every INSEE's endpoints.

As a side effect, this impacts the API's consumption rate, as the 429 error is rarely encountered (for instance on metadata API you may easily end up with a 4/5 queries / second on the average).

**Comment by hadrilec** [comment]
indeed, I think we should include this feature back into the package. @tgrandje would you mind making the necessary changes to that tests pass?

**Comment by tgrandje** [comment]
I'll try to run the tests locally (I think I had one failure I hadn't the time to debug last week): I'm still dumbfounded about those failures on github...

**Comment by tgrandje** [comment]
@hadrilec locally (python 3.10) I have only one failed test, and that had already been covered by @tfardet [here](https://github.com/InseeFrLab/pynsee/pull/220). Do you want me to test locally on all different python versions?

*(Saddly enough, api rate consumption doesn't seem to explain why tests are freezing on github...)*

**Comment by hadrilec** [comment]
alright, let's merge then?
